core/include/mutex.h

```diff
@@ -249,7 +249,24 @@ static inline int mutex_trylock(mutex_t *mutex)
  *
  * @post    The mutex @p is locked and held by the calling thread.
  */
+#if (MAXTHREADS > 1) || DOXYGEN
 void mutex_lock(mutex_t *mutex);
+#else
+/**
+ * @brief   dummy implementation for when no scheduler is used
+ */
+static inline void mutex_lock(mutex_t *mutex)
+{
+    /* (ab)use next pointer as lock variable */
+    volatile uintptr_t *lock = (void *)&mutex->queue.next;
+
+    /* spin until lock is released (e.g. by interrupt) */
+    while (*lock) {}
```

Comment:
```suggestion
    /* spin until lock is released (e.g. by interrupt).
     * Note: since only the numbers 0 and 1 are ever stored in lock, this
     * read does not need to be atomic here - even while a concurrent write
     * is performed on lock, a read will still either yield 0 or 1 (so the old
     * or new value, which both is fine), even if the lock is read out byte-wise
     * (e.g. on AVR).
     */
    while (*lock) {}
```

I'm not insisting, but maybe it makes sense to document why only `volatile` and not an atomic read is used here.